File: Data Structures & Algorithms/search-for-word/submission-0.cs
public class Solution {
    private int ROWS, COLS;
    
    public bool Exist(char[][] board, string word) {

        ROWS = board.Length;
        COLS = board[0].Length;

        for(int r = 0; r < ROWS; r++){
            for(int c = 0; c < COLS; c++){
                if(Dfs(board, word, r, c, 0)){
                    return true;
                }
            }
        }
        
        return false;
    }

    private bool Dfs(char[][] board, string word, int r, int c, int i){
        //means every character from index 0 up through word.Length - 1 
        //has already been matched successfully. 
        if(i == word.Length) {
            return true;
        }
 
        if(r < 0 || c < 0 || r >= ROWS || c >= COLS 
        || board[r][c] != word[i] || board[r][c] == '#'){
            return false; 
        }

        board[r][c] = '#'; // mark current cell as used
                bool res = Dfs(board, word, r + 1, c, i + 1) ||
                   Dfs(board, word, r - 1, c, i + 1) ||
                   Dfs(board, word, r, c + 1, i + 1) ||
                   Dfs(board, word, r, c - 1, i + 1);
        board[r][c] = word[i];
        return res;

    }
}

File: Data Structures & Algorithms/valid-sudoku/submission-0.js
class Solution {
    /**
     * @param {character[][]} board
     * @return {boolean}
     */
    isValidSudoku(board) {
        // rows
        for (let i = 0; i < board.length; i++) {
            let set = new Set();
            for (let j = 0; j < board[i].length; j++) {
                const val = board[i][j];
                if (val === ".") continue;
                if (set.has(val)) return false;
                set.add(val);
            }
        }

        // columns
        for (let j = 0; j < board[0].length; j++) {
            let set = new Set();
            for (let i = 0; i < board.length; i++) {
                const val = board[i][j];
                if (val === ".") continue;
                if (set.has(val)) return false;
                set.add(val);
            }
        }

        // 3x3 boxes
        const start = [0, 3, 6];
        for (let r of start) {
            for (let c of start) {
                let set = new Set();
                for (let i = 0; i < 3; i++) {
                    for (let j = 0; j < 3; j++) {
                        const val = board[r + i][c + j];
                        if (val === ".") continue;
                        if (set.has(val)) return false;
                        set.add(val);
                    }
                }
            }
        }

        return true;
    }
}
